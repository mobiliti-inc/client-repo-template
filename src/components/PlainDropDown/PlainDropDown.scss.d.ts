// This file is automatically generated.
// Please do not change this file!
interface CssExports {
  'dropdown-menu': string;
  'dropdown-menu__bottom': string;
  'dropdown-menu__top': string;
  'dropdown-menu__top-close': string;
  'dropdown-menu__top-open': string;
  'dropdown-menu__top-open--divider': string;
  'fades-in-out': string;
}
declare var cssExports: CssExports;
export = cssExports;
