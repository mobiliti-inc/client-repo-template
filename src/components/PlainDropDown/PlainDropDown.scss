@import '../../styles/variables';

.dropdown-menu {
    color: $white;
    font-family: $futura;
    font-style: normal;
    font-weight: 500;
    font-size: 14px;
    line-height: 19px;
    cursor: pointer;
    text-transform: capitalize;
    &__top {
        display: grid;
        border-bottom: 1px solid transparent;
        transition: all 0.5s ease-in-out;
        -webkit-transition: all 0.5s ease-in-out;

        &-open {
            display: grid;
            &+.dropdown-menu__bottom {
                visibility: visible;
                opacity: 1;
                height: 100px;
            }
            &--divider {
                border-bottom: 1px solid rgba(0, 0, 0, 0.123);
            }
        }

        &-close {
            display: grid;
            &+.dropdown-menu__bottom {
                visibility: hidden;
                opacity: 0;
                height: 0;
            }
        }
    }

    &__bottom {
        visibility: hidden;
        opacity: 0;
        height: 0;
        transition: all 0.5s ease-in-out;
        -webkit-transition: all 0.5s ease-in-out;
    }
}




