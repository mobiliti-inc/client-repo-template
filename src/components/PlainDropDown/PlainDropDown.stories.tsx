import { storiesOf } from "@storybook/react";
import { boolean, object } from "@storybook/addon-knobs";
import React, { useState} from "react";
import { PlainDropDown, MenuItem } from "../";
import { dashboard, inventory } from '../../assets';

const topContent = <div style={{height: '40px', display: 'flex', alignItems: 'center'}}></div>;

const  bottomContent = () => {
	const [isActive, setIsActive] = useState(false);
	return (<MenuItem style={{'padding': '0 35px'}} isChild isActive={isActive} onItemClicked={() => setIsActive( prev => !prev)} isIconUp={isActive} menuItemIcon={inventory} menuItemText="Inventory" />);
};

const topContentTwo = () => {
	const [isActive, setIsActive] = useState(false);
	return (<MenuItem  showRightIcon isActive={isActive} onItemClicked={() => setIsActive( prev => !prev)} isIconUp={isActive} menuItemIcon={dashboard} menuItemText="Dashboard" />);
};

storiesOf("PlainDropDown", module)
	.add("No content", () => {
		return(<PlainDropDown
			showContent={boolean("showContent", false)}
			updateShowContent={()=> {}}
			topContent={topContent}
			isSplitted={boolean("isSplitted", false)}
			style={ object('style', { background: '#4996ae' })}/>);
		})
	.add("With Menu Item", () =>
			<PlainDropDown
			showContent={boolean("showContent", false)}
			updateShowContent={()=> {}}
			topContent={topContentTwo()}
			isSplitted={boolean("isSplitted", false)}
			style={ object('style', { background: '#4996ae' })}>
				<div style={{display: 'flex', justifyContent: 'center'}}>
				{bottomContent()}
				</div>
			</PlainDropDown>
		);
