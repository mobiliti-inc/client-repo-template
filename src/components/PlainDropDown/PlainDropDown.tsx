import React, { FC, ReactNode, useState, useEffect } from 'react';
import classNames from 'classnames';

import './PlainDropDown.scss';

type customStyle = {
	background?: string;
	color?: string;
	height?: string;
	width?: string;
};

interface IProps {
	style?: customStyle;
	children?: ReactNode;
	showContent: boolean;
	isSplitted?: boolean;
	topContent?: ReactNode;
	updateShowContent: (showContent: boolean) => void;
}

const PlainDropDown: FC<IProps> = ({ style, children, topContent, showContent, updateShowContent, isSplitted }) => {
	const [isDroppedDown, setIsDroppedDown] = useState(showContent);

	const handleClick = () => {
		setIsDroppedDown(!isDroppedDown);
		updateShowContent(!isDroppedDown);
		console.log('aa');
	};
	const dropDownMenuClass = classNames('dropdown-menu__top',
		{ 'dropdown-menu__top-open': isDroppedDown },
		{ 'dropdown-menu__top-close': !isDroppedDown },
		{ 'dropdown-menu__top-open--divider': isSplitted && isDroppedDown }
	);

	useEffect(() => {
		setIsDroppedDown(showContent);
	}, [showContent]);
	return (
		<div
		style={style}
		styleName="dropdown-menu"
		>
				<div
				role="presentation"
				styleName={dropDownMenuClass}
				onClick={handleClick}>
					{topContent}
				</div>
				<div styleName="dropdown-menu__bottom">
					{children}
				</div>
		</div>
	);
};

export default PlainDropDown;