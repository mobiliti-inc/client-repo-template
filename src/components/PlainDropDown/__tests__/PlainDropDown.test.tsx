import React from 'react';
import { act } from 'react-dom/test-utils';

import PlainDropDown from '../PlainDropDown';

describe('PlainDropDown', () => {
	it('renders with content not shown', () => {
		const subject = shallow(<PlainDropDown showContent={false} updateShowContent={jest.fn()} />);
		expect(subject).toMatchSnapshot();
	});

	it('renders with content shown', () => {
		const subject = shallow(<PlainDropDown showContent updateShowContent={jest.fn()} />);
		expect(subject).toMatchSnapshot();
	});

	it('renders with horizontal line between top and bottom content', () => {
		const subject = shallow(<PlainDropDown isSplitted showContent updateShowContent={jest.fn()} />);
		expect(subject).toMatchSnapshot();
	});

	it('drops down when clicked', () => {
		const subject = mount(<PlainDropDown isSplitted showContent={false} updateShowContent={jest.fn()} />);
		const PlainDropDownTopDiv =subject.find('.dropdown-menu__top');
		expect(PlainDropDownTopDiv).toHaveLength(1);
		expect(subject.find('.dropdown-menu__top-open')).toHaveLength(0);

		act(() => {
			PlainDropDownTopDiv.simulate('click');
		});
		subject.update();
		expect(subject).toMatchSnapshot();
		expect(subject.find('.dropdown-menu__top-open')).toHaveLength(1);
	});
});
