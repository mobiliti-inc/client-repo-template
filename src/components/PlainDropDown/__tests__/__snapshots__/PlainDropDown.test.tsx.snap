// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlainDropDown drops down when clicked 1`] = `
<PlainDropDown
  isSplitted={true}
  showContent={false}
  updateShowContent={
    [MockFunction] {
      "calls": Array [
        Array [
          true,
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <div
    className="dropdown-menu"
  >
    <div
      className="dropdown-menu__top dropdown-menu__top-open dropdown-menu__top-open--divider"
      onClick={[Function]}
      role="presentation"
    />
    <div
      className="dropdown-menu__bottom"
    />
  </div>
</PlainDropDown>
`;

exports[`PlainDropDown renders with content not shown 1`] = `
<div
  className="dropdown-menu"
>
  <div
    className="dropdown-menu__top dropdown-menu__top-close"
    onClick={[Function]}
    role="presentation"
  />
  <div
    className="dropdown-menu__bottom"
  />
</div>
`;

exports[`PlainDropDown renders with content shown 1`] = `
<div
  className="dropdown-menu"
>
  <div
    className="dropdown-menu__top dropdown-menu__top-open"
    onClick={[Function]}
    role="presentation"
  />
  <div
    className="dropdown-menu__bottom"
  />
</div>
`;

exports[`PlainDropDown renders with horizontal line between top and bottom content 1`] = `
<div
  className="dropdown-menu"
>
  <div
    className="dropdown-menu__top dropdown-menu__top-open dropdown-menu__top-open--divider"
    onClick={[Function]}
    role="presentation"
  />
  <div
    className="dropdown-menu__bottom"
  />
</div>
`;
